### PR TITLE
#1045 Fix flaky BasicTests4Collie

### PR DESCRIPTION
**Pull request by chia7712**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #1045)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

**Comment by chia7712** [comment]
retry it

**Comment by chia7712** [comment]
retry configurator

**Comment by chia7712** [comment]
@jackyoh 能否開個議題看一下k8s環境是不是又在調皮了

**Comment by jackyoh** [comment]
> @jackyoh 能否開個議題看一下k8s環境是不是又在調皮了
ohara it 有一台 k8s 的 slave node 因為記憶體使用過多而當掉. 我有把那台 node 重開, 等等我會來測試看看.

File: docs/rest_interface.md [hunk 1]
```diff
@@ -1584,18 +1584,12 @@ It is disallowed to remove a zookeeper cluster used by a running [broker cluster
...
-  "nodeNames": [
-    "node00"
-  ]
-}
 ```
+402 NoContent
+```
+
+> It is ok to delete an nonexistent zookeeper cluster, and the response is 402 NoContent.
+
```
Comment (saivirtue) [inline docs/rest_interface.md]: Is this related to #1039 ?

File: docs/rest_interface.md [hunk 2]
```diff
@@ -1746,20 +1752,14 @@ It is disallowed to remove a broker cluster used by a running [worker cluster](#
...
-  ]
-}
 ```
+402 NoContent
+```
+
+> It is ok to delete an nonexistent broker cluster, and the response is 402 NoContent.
+
 ----------
+
```
Comment (saivirtue) [inline docs/rest_interface.md]: ditto

File: ohara-agent/src/main/scala/com/island/ohara/agent/BrokerCollie.scala [hunk 3]
```diff
@@ -79,6 +65,19 @@ object BrokerCollie {
...
+        zookeeperClusterName(bk.zookeeperClusterName)
+        clientPort(bk.clientPort)
+        exporterPort(bk.exporterPort)
+        jmxPort(bk.jmxPort)
+        this
+      case _ =>
+        super.copy(clusterInfo)
+        this
+    }
+
```
Comment (saivirtue) [inline ohara-agent/src/main/scala/com/island/ohara/agent/BrokerCollie.scala]: Since this is in a subclass, should we throw an exception instead of use super.copy in the `others` case?

File: docs/rest_interface.md (hunk 1 above)
Comment (chia7712) [reply docs/rest_interface.md]: yep. I had updated the description of #1039

File: ohara-agent/src/main/scala/com/island/ohara/agent/BrokerCollie.scala (hunk 3 above)
Comment (chia7712) [reply ohara-agent/src/main/scala/com/island/ohara/agent/BrokerCollie.scala]: sure

**Comment by jackyoh** [comment]
retry it

**Comment by chia7712** [comment]
retry common
